### PR TITLE
Manage authorities

### LINKED ISSUE #777: Profile Permissions page expand
Expand permissions page to show authorized accounts and simplify private key viewing.
I want us to make it similar to hivesigner.com/auths page, viewing private key section should be similar.

### PR DESCRIPTION
**Pull request by MTSaeed**
**What does this PR?**
Show the option to manage the authorities of current account.
Click on Revoke will ask you to enter the key. after putting right key, you can revoke the permissions.
Click on "Copy" is responsible to copy the key. click on "Import private key" will ask you to enter master password or that specific key . 
When you enters "Master password" it will reveal all the private keys to you. When you enters that specific key like owner, 
 active or posting key, it will just reveal that key.
When user puts Master password at the time of logging, it will generate private keys and store in the Local Storage. When user 
 visits profile permissions page, it will get that keys from local storage and shows that key.
When user login by putting Private keys like active or posting key, it will store that key in local Storage and reveal that specific 
 key in the profile permission page. 
When user login in ecency by using any other 3rd party website like hiveSigner etc and visit the profile permission page, it will 
 show you the option to import private keys. you can see this scenario in the below video.
Shows another custom component when screen size is less than sm-breakpoint.

**Steps to reproduce**

1. Go to profile page. --> Settings  --> Permissions.
2. Click on "Revoke" and enter sign key. you can perform this transaction by Hive signer and by key chain too.
3. Click on "Copy" button to copy that key. 
4. Click on "Import private key" and if you put Master password, it will reveal all the private keys. But when you enter that 
    specific key , it will just reveal that key.
5. You can watch below video for more guidance.

**Issue number** 

fixes #777

**Screenshot / Video**

### CODE REVIEW

**Comment by MTSaeed** [comment]
image

**Comment by MTSaeed** [comment]
https://user-images.githubusercontent.com/106739598/224735260-8328d0e5-128a-4f4a-92b5-b7907728544d.mp4

**Comment by MTSaeed** [comment]
https://user-images.githubusercontent.com/106739598/224741324-a5d160ff-f904-494d-88f2-1957d58e7085.mp4

**Comment by MTSaeed** [comment]
https://user-images.githubusercontent.com/106739598/224904762-25688558-ef7c-43c9-9bda-4283631e6199.mp4

**Comment by feruzm** [comment]
@MTSaeed From your video I see that, Dropdown has styling issue, there is unnecessary top padding. Also dialog which imports key says transaction successful after import which is not good, because it is not transaction 😄 
I will do some functionality tests and review code.

**Comment by MTSaeed** [comment]
> @MTSaeed From your video I see that, Dropdown has styling issue, there is unnecessary top padding. Also dialog which imports key says transaction successful after import which is not good, because it is not transaction smile I will do some functionality tests and review code.

Hi @feruzm 
Could you please mention from which you video you have noticed that the Dropdown has styling issue?. I will fix that issue.
Please suggest the suitable title for that dialog, i will update the title.

File: src/common/components/feedback/_index.scss [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 .feedback-container {
   position: fixed;
-  z-index: 99;
+  z-index: 999;
```
Comment (dkildar) [inline src/common/components/feedback/_index.scss:19]: Why it had been done? We don't use multiplying of numbers. Use +1 instead. 100 in this case

File: src/common/components/manage-auth-icon/index.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  label?: string;
+  action?: string;
+  keyType?: string;
+  onRevoke?: (account: string) => void;
+  onCopy?: (text: string) => void;
+  onImport?: (type: string) => void;
+  onReveal?: () => void;
+}
+
+export class ManageAuthIcon extends Component<Props> {
```
Comment (dkildar) [inline src/common/components/manage-auth-icon/index.tsx:26]: Please, make is as functional component

File: src/common/components/manage-authority/index.tsx [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+  global: Global;
+  activeUser: ActiveUser | null;
+  signingKey: string;
+  setSigningKey: (key: string) => void;
+  history: History;
+}
+
+export default function ManageAuthorities(props: Props) {
+  const [accountData, setAccountdata] = useState<AccountDataType>();
+  const [newPostingsAuthority, setNewPostingsAuthority] = useState<Array<any>>([]);
```
Comment (dkildar) [inline src/common/components/manage-authority/index.tsx:34]: Why its any?

File: src/common/components/manage-authority/index.tsx [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+      <>
+        <div className="sign-dialog-header border-bottom">
+          <div className="step-no">1</div>
+          <div className="sign-dialog-titles">
+            <div className="authority-main-title">{_t("manage-authorities.sign-title")}</div>
+            <div className="authority-sub-title">{_t("manage-authorities.sign-sub-title")}</div>
+          </div>
+        </div>
+        {inProgress && <LinearProgress />}
+        {keyOrHot({
```
Comment (dkildar) [inline src/common/components/manage-authority/index.tsx:136]: Please, declare components as element: `<KeyOrHost ... />`

File: src/common/components/manage-authority/types/manage-authorities.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  Active: "active",
+  Posting: "posting",
+  Memo: "memo"
+};
+
+export interface PublicKeys {
+  publicOwnerKey?: any;
+  publicActiveKey?: any;
+  publicPostingKey?: any;
+  publicMemoKey?: any;
```
Comment (dkildar) [inline src/common/components/manage-authority/types/manage-authorities.ts:14]: Please, specify types

File: src/common/components/manage-authority/types/manage-authorities.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+export interface AccountDataType {
+  postingsAuthority: Array<any>;
+  posting: Array<any>;
+  owner: Array<any>;
+  active: Array<any>;
```
Comment (dkildar) [inline src/common/components/manage-authority/types/manage-authorities.ts:21]: 1. just `any[]`;
2. please, specify types;

File: src/common/components/manage-authority/types/manage-authorities.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  weight: number;
+  memokey: string;
+  account: Account;
+  PublicKeys: {
+    publicOwnerKey: any;
+    publicActiveKey: any;
+    publicPostingKey: any;
+    publicMemoKey: any;
```
Comment (dkildar) [inline src/common/components/manage-authority/types/manage-authorities.ts:29]: please, specify types

File: src/common/components/view-keys/index.tsx [hunk 8]
```diff
@@ -1,21 +1,18 @@
...
 import BaseComponent from "../base";
-import { error, success } from "../feedback";
-
-import { formatError } from "../../api/operations";
-
+import { success } from "../feedback";
+import { generateKeys } from "../../helper/generate-private-keys";
 import { _t } from "../../i18n";
 
+import _ from "lodash";
```
Comment (dkildar) [inline src/common/components/view-keys/index.tsx:11]: Please, avoid lodash because we want to get rid of it

**Comment by MTSaeed** [comment]
image

**Comment by MTSaeed** [comment]
Issue:
image

Fixed:
image

File: src/common/components/manage-auth-icon/index.tsx [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+      }
+    }
+  };
+
+  let menuItems: MenuItem[] = [];
+  switch (props.type) {
+    case actionType.Revoke:
+      menuItems = [
+        {
+          label: "Revoke",
```
Comment (feruzm) [inline src/common/components/manage-auth-icon/index.tsx:56]: @MTSaeed these labels should be translatable if part of UI.

File: src/common/components/manage-authority/index.tsx [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+      const resp = response[0];
+      setAccountdata({
+        postingsAuthority: resp.posting.account_auths,
+        posting: resp.posting.key_auths[0],
+        owner: resp.owner.key_auths[0],
+        active: resp.active.key_auths[0],
+        weight: resp.active.weight_threshold,
+        memokey: resp.memo_key,
+        account: response[0],
+        PublicKeys: {
```
Comment (feruzm) [inline src/common/components/manage-authority/index.tsx:56]: No need uppercase, just `publicKeys`

File: src/common/components/manage-auth-icon/index.tsx (hunk 9 above)
Comment (MTSaeed) [reply src/common/components/manage-auth-icon/index.tsx]: @feruzm Updated.

File: src/common/components/manage-authority/index.tsx (hunk 10 above)
Comment (MTSaeed) [reply src/common/components/manage-authority/index.tsx]: updated.

**Pull request merged**